### PR TITLE
Improved plugin performance.

### PR DESCRIPTION
**Pull request by dy0nysos**
No more useless data stored and no more big array manipulated.

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
Hi @dy0nysos,

Thanks for your PR, I let @s4rbacane having a look on it.

Best regards

**Comment by s4rbacane** [comment]
Hi, you can accept all pull requests from @dy0nysos .

**Comment by Quetzacoalt91** [comment]
Thanks for the information ;)

File: upgrade/install-1.0.6.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			IF NEW.newsletter = 1 THEN
+				INSERT INTO '._DB_PREFIX_.'sd_updates (id,update_date,customer_id,customer_email,action) VALUES (NULL,NOW(),NEW.id_customer,NEW.email,\'S\');
+			ELSE
+				INSERT INTO '._DB_PREFIX_.'sd_updates (id,update_date,customer_id,customer_email,action) VALUES (NULL,NOW(),NEW.id_customer,NEW.email,\'U\');
+			END IF;
+		END;
+		');
+	$update4 = Db::getInstance()->execute('
+		ALTER TABLE `'._DB_PREFIX_.'sarbacanedesktop_users`
+		ADD COLUMN `last_call_date` DATETIME NULL AFTER `list_id`;
```
Comment (Quetzacoalt91) [inline upgrade/install-1.0.6.php]: The column `list_id` seems to be missing. I guess the upgrade won't work.

Comment (dy0nysos) [inline upgrade/install-1.0.6.php]: I just fixed it.
I put this column in the install method of the main class but forgot to put it here.

File: sarbacanedesktop.php [hunk 2]
```diff
@@ -837,5 +677,10 @@ public function getContent()
 		));
 		return $this->context->smarty->fetch($this->local_path.'views/templates/admin/sarbacanedesktop.tpl');
 	}
+	
+	function clearHistory(){
+		$rq_sql = 'DELETE FROM `'._DB_PREFIX_.'sd_updates` WHERE update_date<= (SELECT MIN(last_call_date) FROM `'._DB_PREFIX_.'sarbacanedesktop_users`)';
```
Comment (Quetzacoalt91) [inline sarbacanedesktop.php]: This function is not really useful, because the request is not executed or returned.

File: sarbacanedesktop.php [hunk 3]
```diff
@@ -544,52 +472,20 @@ private function processNewUnsubscribers($list_type, $id_shop, $id_sd_id, $type_
...
-				SELECT c.`email`
-				FROM `'._DB_PREFIX_.'customer` c
-				WHERE c.'.$shop_customer_selection.'
-				AND c.`deleted` = 0
-			)
-			AND s.`list_type` = \'C\'
-			AND s.`id_shop` = \''.pSql($id_shop).'\'
-			AND s.`id_sd_id` = \''.pSql($id_sd_id).'\'
-			'.$rq_sql_limit;
+			SELECT * FROM '._DB_PREFIX_.'customer WHERE 1=2';
```
Comment (Quetzacoalt91) [inline sarbacanedesktop.php]: I don't understand why you write this SQL request. You can return directly if you get this value

**Comment by Quetzacoalt91** [comment]
Hi @dy0nysos & @s4rbacane,

Thanks, the PR has just been merged and the changes should be available from tomorrow.

Best regards

**Comment by dy0nysos** [comment]
Hi @Quetzacoalt91,
Thanks you for your involvement and your code review.
It was really helpful.

**Pull request merged**